### PR TITLE
Minor refactor recycled from closed PR

### PR DESCRIPTION
**Pull request by dmos62**
This is a small collection of refactors recycled from a rejected PR (https://github.com/centerofci/mathesar/pull/1148).

It's also a minor extension of the filters endpoint to show some information relevant for equality-based filters. This will become useful again when we come back to the issue addressed by #1148. No breaking changes.

Some changes are only formatting, because, for example, I added some parameters to a function in the PR this is based on, making the declaration multi-line, but then removed the new parameters for this PR, after which only formatting changes were left. I don't think any of the formatting changes are detrimental on their own, so I didn't bother undoing them. 

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: mathesar/api/db/viewsets/records.py [hunk 1]
```diff
@@ -36,27 +36,28 @@ def list(self, request, table_pk=None):
...
         if grouping:
             group_by_columns_names = [column_ids_to_names[column_id] for column_id in grouping['columns']]
             name_converted_group_by = {**grouping, 'columns': group_by_columns_names}
         name_converted_order_by = [{**column, 'field': column_ids_to_names[column['field']]} for column in order_by]
-
         try:
-
+            filter_processed = _rewrite_db_function_spec(
+                filter_unprocessed, column_ids_to_names,
+            )
```
Comment (mathemancer) [inline mathesar/api/db/viewsets/records.py:52]: I think this logic would fit much better in the models utility module, `mathesar.utils.models`.

**Review by mathemancer — CHANGES_REQUESTED** [review]
Looks pretty good overall. I think we should move the column id to name wrangling out of the viewset, and down into a model utility, since it's more related to the models (and so that would be easier to find/understand later).

**Comment by dmos62** [comment]
@mathemancer in reply to:

[> I think this logic would fit much better in the models utility module, `mathesar.utils.models`.](https://github.com/centerofci/mathesar/pull/1174#discussion_r827653822):

The actual logic that does the rewriting is actually in the `mathesar.functions.operations.convert` namespace. This method in the records viewset file is just a wrapped call to that, which also adds handling for input being None, and includes a (hopefully helpful) doc string and some comments, so I didn't move it to another namespace.

[I removed this method, unwrapping the call to the rewriting method.](https://github.com/centerofci/mathesar/pull/1174/commits/4b82f5a70ce1bb5d587fa0f4894a4086d0e2b51f) I think it was better before. If you agree, we can revert back.

----------
@mathemancer in reply to:

[> Since you've set things up with default `None`s in the function sig, this conditional isn't needed: `limit(self, limit)` is a no-op when `limit = None`.](https://github.com/centerofci/mathesar/pull/1174#discussion_r827645339)

I'm aware that some of the `None` checks are redundant. But, the idea here is to treat each transformation in this pipeline the same. I tried to underline their conceptual equality by wrapping each of them in an identical `if` block, even if it's redundant in a few cases. I wanted to make it obvious that this is ripe for some refactoring: maybe something that would make the pipish nature of this routine more explicit.

Can we keep the redundant `if` checks?

**Comment by dmos62** [comment]
@kgodey in this particular case, it's more confusing to have something not `None`-checked. I think a reader would take a second to realise that the SA call probably does a no-op if it's passed a `None`, but he can't be sure, because the enclosing method might never get called with that parameter set to `None`. These "redundant" if blocks tell the reader "there's nothing to think about: sometimes this is `None`, and in that case this block doesn't happen".

Some comments might be a good idea, but some code structure can be worth a lot of comments.

**Comment by mathemancer** [comment]
> @mathemancer in reply to:
> 
> [> I think this logic would fit much better in the models utility module, `mathesar.utils.models`.](https://github.com/centerofci/mathesar/pull/1174#discussion_r827653822)
> 
> The actual logic that does the rewriting is actually in the `mathesar.functions.operations.convert` namespace. This method in the records viewset file is just a wrapped call to that, which also adds handling for input being None, and includes a (hopefully helpful) doc string and some comments, so I didn't move it to another namespace.
>
> [I removed this method, unwrapping the call to the rewriting method.](https://github.com/centerofci/mathesar/pull/1174/commits/4b82f5a70ce1bb5d587fa0f4894a4086d0e2b51f) I think it was better before. If you agree, we can revert back.

I strongly prefer the separate function version to inline. My problem is having non-exception branching in a viewset at all. That said, as I look at our other viewset code, maybe I'm in the minority there. I'm okay with reverting to the wrapped version, getting that in, and reassessing where we want what logic later.

> @mathemancer in reply to:
> 
> [> Since you've set things up with default `None`s in the function sig, this conditional isn't needed: `limit(self, limit)` is a no-op when `limit = None`.](https://github.com/centerofci/mathesar/pull/1174#discussion_r827645339)
> 
> I'm aware that some of the `None` checks are redundant. But, the idea here is to treat each transformation in this pipeline the same. I tried to underline their conceptual equality by wrapping each of them in an identical `if` block, even if it's redundant in a few cases. I wanted to make it obvious that this is ripe for some refactoring: maybe something that would make the pipish nature of this routine more explicit.
> 
> Can we keep the redundant `if` checks?

I think this will be a problem, since different pipeline bits might treat `None` differently. There might be cases where some explicit `None` is used. Further, even in the case we have, this added complexity adds the potential for bugs, and these are more complicated to test for in this setting than they would be function-by-function. `limit` and `offset` treat the falsy `0` value differently in comparison to the falsy `None`. A query with a limit of `0` can be used to return just column names (here, that possibility is short-circuited), whereas the offset of `0` is equal to an offset of `None` (with our current logic). While this specific example can be sorted by changing all checks to `if foo is None:` rather than `if foo:`, my broader point is that what to do about a `None` input should belong to the bit of the pipeline rather than to the caller, and tested for each part separately. In the cases of functions in the pipeline that _don't_ do the right thing when handed `None`, I think we should instead get those sorted out bit by bit.

**Comment by dmos62** [comment]
This PR will likely be closed, since it's being superseded by https://github.com/centerofci/mathesar/pull/1222, but I'm keeping it open until after I merge the other one, since they might not completely overlap.

**Pull request closed without merging**